Source/WarpX.cpp

```diff
@@ -549,6 +562,45 @@ WarpX::ReadParameters ()
             }
         }
 
+        std::vector<std::string> signals_in;
+        pp_warpx.queryarr("break_signals", signals_in);
+        for (const std::string &str : signals_in) {
+            int sig = parseSignalNameToNumber(str);
+            AMREX_ALWAYS_ASSERT(sig < NUM_SIGNALS);
+            signal_conf_requests_break[sig] = true;
+        }
+        signals_in.clear();
+        pp_warpx.queryarr("checkpoint_signals", signals_in);
+        for (const std::string &str : signals_in) {
+            int sig = parseSignalNameToNumber(str);
+            AMREX_ALWAYS_ASSERT(sig < NUM_SIGNALS);
+            signal_conf_requests_checkpoint[sig] = true;
+        }
+#if defined(__GNU_LIBRARY__) || defined(__APPLE__)
```

Comment:
This may miss covering Linux builds using musl libc or the like. I'm willing to bet there are other elements in the stack that rely on GNU libc functionality, though.